chanzuckerberg/single-cell-data-portal
# fix: 5936 cell type share link auto expand

## Reason for Change

- #5936 

## Changes

1. CellTypes in share link will now auto expand tissues
2. BONUS: Fix extra bug that cellTypes in share link doesn't result in cellTypeIds in `/filters` API request
3. Add corresponding tests to cover both bugs above
4. EXTRA BONUS: Refactor Heatmap component to use Ronen Pattern™️ 

## Testing steps

1. Visit https://pr-6050-frontend.rdev.single-cell.czi.technology/gene-expression?genes=TSPAN6%2CTNMD%2CDPM1&ver=2&cellTypes=B+cell%2CB-1a+B+cell
2. The tissues should be expanded like so:

<img width="1064" alt="Screenshot 2023-10-20 at 8 41 53 AM" src="https://github.com/chanzuckerberg/single-cell-data-portal/assets/6309723/dbe5413e-bb59-4aef-9bae-d11740c6a97b">

3. API call should include cellTypeIds too

<img width="1554" alt="Screenshot 2023-10-20 at 8 42 36 AM" src="https://github.com/chanzuckerberg/single-cell-data-portal/assets/6309723/503bed51-5864-4d27-b530-bcf279ce8001">


## Notes for Reviewer
